<a href="https://colab.research.google.com/github/sinman96/Data-Science-Projects/blob/main/Monty_Hall_Problem.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [52]:
import numpy as np
import random

**I will label the doors 1,2,3**

**Function outputting probability when you don't switch doors for 10000 simulations.**

In [53]:
def keep_door(chosen_door): 
    success_count = 0
    for i in range(0, 10000):
        doors = np.array([1,2,3])
        prize_door = random.choices(doors)[0]
        """As your selection is fixed,
        there's no need for further filtering"""
        open_door = random.choices(doors[(doors != chosen_door)
        & (doors != prize_door)])[0]
        #As the first door has been opened we can discard this.
        doors = np.delete(doors, np.where(doors == open_door))
        if chosen_door == prize_door:                
           success_count += 1        
    return success_count/10000

**Function outputting probability when you switch doors for 10000 simulations.**

In [54]:
def change_door(chosen_door): 
    success_count = 0
    for i in range(0, 10000):
        doors = np.array([1,2,3])
        prize_door = random.choices(doors)[0]
        open_door = random.choices(doors[(doors != chosen_door)
        & (doors != prize_door)])[0]
        #As the first door has been opened we can discard this.
        doors = np.delete(doors, np.where(doors == open_door))
        #This is switching the choice of door.
        chosen_door = doors[(doors != chosen_door)]
        if chosen_door == prize_door:
               success_count += 1
    return success_count/10000

**Different probabilities for each door.**

In [55]:
keep_door(1), change_door(1)

(0.3349, 0.6732)

In [56]:
keep_door(2), change_door(2)

(0.3386, 0.6668)

In [57]:
keep_door(3), change_door(3)

(0.3366, 0.6621)

**Averages of probabilities.**

In [58]:
(keep_door(1) + keep_door(2) + keep_door(3))/3, (change_door(1) + change_door(2) + change_door(3))/3

(0.33413333333333334, 0.6696)

**The average probabilities of sticking and switching doors to 2 decimal places are (0.33, 0.67). This gives strong evidence that p_stick = 1/3 and p_switch = 2/3.**